frontend/src/metabase/visualizations/lib/settings.js

```diff
@@ -287,3 +287,26 @@ export function mergeSettings(first: Settings = {}, second: Settings = {}) {
   }
   return merged;
 }
+
+export function getClickBehaviorSettings(settings) {
+  const newSettings = {};
+
+  if (settings.click_behavior) {
+    newSettings.click_behavior = settings.click_behavior;
+  }
+
+  if (settings.column_settings) {
+    Object.entries(settings.column_settings).forEach(([key, fieldSettings]) => {
+      if (fieldSettings.click_behavior == null) {
+        return;
+      }
+
+      newSettings.column_settings = newSettings.column_settings || {};
+      newSettings.column_settings[key] = {
+        click_behavior: fieldSettings.click_behavior,
+      };
+    });
+  }
```

Comment:
This block _could_ be extracted onto a named function.

Benefits might be:

1. giving the function a nice name that will help readers grasp the intent of this block quickly.
2. making function `getClickBehaviorSettings` very quick to understand overall.

`newSettings` could be mutated in this other function and not even need to be returned explicitly from said function for everything to work fine.

Reply:
…in other words, less cyclomatic complexity.